### PR TITLE
chore: format and cleanup test scripts

### PR DESCRIPTION
**Pull request by frezbo**
This formats and cleanups the test scripts.

### CODE REVIEW

File: hack/test/e2e.sh [hunk 1]
```diff
@@ -49,127 +51,126 @@ PROVISIONER=
...
 
 # Create a cluster via CAPI.
 function create_cluster_capi {
   trap cleanup_capi EXIT
 
-  ${KUBECTL} --kubeconfig /tmp/e2e/docker/kubeconfig apply -f ${TMP}/cluster.yaml
+  ${KUBECTL} --kubeconfig /tmp/e2e/docker/kubeconfig apply -f "${TMP}/cluster.yaml"
 
   # Wait for first controlplane machine to have a name
-  timeout=$(($(date +%s) + ${TIMEOUT}))
```
Comment (smira) [inline hack/test/e2e.sh:62]: this dropped `$` looks weird

Comment (frezbo) [reply hack/test/e2e.sh:62]: that's intentional, no need of interpolation when using the `$((` syntax

Comment (frezbo) [reply hack/test/e2e.sh:62]: https://www.shellcheck.net/wiki/SC2004

**Review by smira — APPROVED** [review]

File: hack/test/e2e-qemu.sh [hunk 2]
```diff
@@ -173,19 +164,12 @@ function create_cluster {
     --cidr=172.20.1.0/24 \
     --user-disk=/var/lib/extra:100MB \
     --user-disk=/var/lib/p1:100MB:/var/lib/p2:100MB \
-    --install-image=${INSTALLER_IMAGE} \
+    --install-image="${INSTALLER_IMAGE}" \
     --with-init-node=false \
-    --cni-bundle-url=${ARTIFACTS}/talosctl-cni-bundle-'${ARCH}'.tar.gz \
+    --cni-bundle-url="${ARTIFACTS}/talosctl-cni-bundle-\${ARCH}.tar.gz" \
```
Comment (DmitriyMV) [inline hack/test/e2e-qemu.sh:169]: I'm not bash expert so is there any reason for `\` before `${ARCH}` here?

**Comment by smira** [comment]
looks like `e2e` is broken

File: hack/test/e2e.sh [hunk 3]
```diff
@@ -198,25 +199,25 @@ function get_kubeconfig {
...
 
 function build_registry_mirrors {
   if [[ "${CI:-false}" == "true" ]]; then
-    REGISTRY_MIRROR_FLAGS=
+    REGISTRY_MIRROR_FLAGS=()
 
     for registry in docker.io registry.k8s.io quay.io gcr.io ghcr.io registry.dev.talos-systems.io; do
       local service="registry-${registry//./-}.ci.svc"
-      local addr=`python3 -c "import socket; print(socket.gethostbyname('${service}'))"`
+      addr=$(python3 -c "import socket; print(socket.gethostbyname('${service}'))")
```
Comment (DmitriyMV) [inline hack/test/e2e.sh:214]: Why drop `local`?

**Comment by frezbo** [comment]
> looks like `e2e` is broken

i think i fixed that now, let's see

**Review by DmitriyMV — APPROVED** [review]
🆒 usage of bash arrays.

File: hack/test/e2e-qemu.sh (hunk 2 above)
Comment (frezbo) [reply hack/test/e2e-qemu.sh:169]: so that it does not expand, the cli expect `${ARCH}` as is and not expanded

File: hack/test/e2e.sh (hunk 3 above)
Comment (frezbo) [reply hack/test/e2e.sh:214]: https://www.shellcheck.net/wiki/SC2155

File: hack/test/e2e-qemu.sh (hunk 2 above)
Comment (DmitriyMV) [reply hack/test/e2e-qemu.sh:169]: Ah! Can't we use single quotes here tho?

File: hack/test/e2e.sh (hunk 3 above)
Comment (DmitriyMV) [reply hack/test/e2e.sh:214]: Maybe declare it separately then?

Comment (frezbo) [reply hack/test/e2e.sh:214]: that didn't matter much here :sweat_smile:

File: hack/test/e2e-qemu.sh (hunk 2 above)
Comment (frezbo) [reply hack/test/e2e-qemu.sh:169]: since the whole expression is inside a double quote, it would still expand

**Pull request merged**